membrane_curvature/surface.py

```diff
@@ -131,3 +131,60 @@ def normalized_grid(grid_z_coordinates, grid_norm_unit):
     z_normalized = grid_z_coordinates / grid_norm_unit
 
     return z_normalized
+
+
+def interpolation_by_array(array_surface):
+    """
+        Interpolates values contained in `array_surface` over axis
+
+    Parameters
+    ----------
+
+    array_surface: np.ndarray
+        Array of floats of shape (`n_x_bins`, `n_y_bins`)
+
+    Returns
+    -------
+    interpolated_array: np.ndarray
+        Returns interpolated array.
+        Array of shape (`n_x_bins` x `n_y_bins`,)
+
+    """
+
+    # create mask for nans
+    mask_nans = ~np.isnan(array_surface)
+
+    # index of array_surface
+    index_array = np.arange(array_surface.shape[0])
+
+    # interpolate values in array
+    interpolated_array = np.interp(index_array,
+                                   np.flatnonzero(mask_nans),
+                                   array_surface[mask_nans])
+
+    return interpolated_array
+
+
+def surface_interpolation(array_surface):
+    """
+        Calculates interpolation of arrays.
+
+    Parameters
+    ----------
+
+    array_surface: np.ndarray
+        Array of floats of shape (`n_x_bins`, `n_y_bins`)
+
+    Returns
+    -------
+
+    interpolated_surface: np.ndarray
+        Interpolated surface derived from set of coordinates
+        in grid of `x_range, y_range` dimensions.
+        Array of floats of shape (`n_x_bins`, `n_y_bins`)
+
+    """
+
+    interpolated_surface = np.apply_along_axis(interpolation_by_array, axis=0, arr=array_surface)
```

Comment:
Ok, I see that you are applying `interpolation_by_array` by row, then. In that case I suggest amending the documentation of `interpolation_by_array` to specify that the input is 1-dimensional with length `n_x`, instead of `n_x, n_y`. Although I think doing 2D interpolation would be less arbitrary (choosing which axis is x and which is y is basically a random, right?)